app/assets/javascripts/Components/assignment_summary_table.jsx

```diff
@@ -1,18 +1,20 @@
 import React from "react";
 import {render} from "react-dom";
-import {markingStateColumn} from "./Helpers/table_helpers";
+import {markingStateColumn, getMarkingStates} from "./Helpers/table_helpers";
 
 import ReactTable from "react-table";
 
 class AssignmentSummaryTable extends React.Component {
   constructor() {
     super();
+    const markingStates = getMarkingStates([]);
     this.state = {
       data: [],
       criteriaColumns: [],
       loading: true,
       num_assigned: 0,
       num_marked: 0,
+      marking_states: markingStates,
```

Comment:
Initialize the state for `markingStateFilter` here as well (to `"all"`). This, plus the `onFilteredChange` comment I made below, should allow you to ensure we never pass `undefined` to `markingStateColumn`, so you can remove that case from the code.